cli/command/image/build.go

```diff
@@ -208,8 +210,12 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 			includes = append(includes, ".dockerignore", relDockerfile)
 		}
 
+		compression := archive.Uncompressed
+		if options.compress {
+			compression = archive.Gzip
```

Comment:
In general you need a very fast compression algorithm to make this worthwhile unless your bandwidth to the docker daemon is very low. Alternatives like LZMA (or even bzip2) are too slow to be worth the overhead in most cases. I originally spec'd this with support for alternate schemes, but removed it as it became clear the complexity wasn't worth the (potential) future advantage.